### PR TITLE
Remove THGeneral

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #69419
* __->__ #69041
* #69040
* #69039
* #69038

`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

Differential Revision: [D32872477](https://our.internmc.facebook.com/intern/diff/D32872477)

### CODE REVIEW

File: aten/src/TH/THConcat.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
```
Comment (ngimel) [inline aten/src/TH/THConcat.h:1]: should this go to torch/csrc, it's used only there?

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32872477).

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
@peterbell10 can you please also create a macro similar to THArgCheck? It's still used a lot internally, and cannot be easily replaced by TORCH_CHECK, because it accepts `(cond, argN, msg...)`

**Comment by peterbell10** [comment]
@ngimel are you asking for `THArgCheck` to be left as it is, or for something like:
```cpp
#define TORCH_CHECK_ARG(cond, argN, ...) TORCH_CHECK(cond,  "invalid argument ", argN, ": ", __VA_ARGS__)
```

**Comment by ngimel** [comment]
`TORCH_CHECK_ARG` is better, it's in line with the rest of the check macros

**Comment by peterbell10** [comment]
@ngimel I've added `TORCH_CHECK_ARG`.

**Comment by ngimel** [comment]
Thanks! I'm trying to land previous diffs in the stack (they required some internal changes)

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32872477).

**Pull request closed without merging**